src/full/Agda/TypeChecking/Sort.hs

```diff
@@ -85,28 +86,39 @@ hasBiggerSort = void . inferUnivSort
 -- | Infer the sort of a pi type. If we can compute the sort straight away,
 --   return that. Otherwise, return @PiSort a s2@ and add a constraint to
 --   ensure we can compute the sort eventually.
-inferPiSort :: PureTCM m => Dom Type -> Abs Sort -> m Sort
+inferPiSort :: (PureTCM m,MonadConstraint m) => Dom Type -> Abs Sort -> m Sort
 inferPiSort a s2 = do
   s1' <- reduce $ getSort a
-  s2' <- mapAbstraction a reduce s2
+  s2' <- reduce s2
+  s2' <- mapAbstraction a reduce s2'
   -- we do instantiateFull here to perhaps remove some (flexible)
   -- dependencies of s2 on var 0, thus allowing piSort' to reduce
   s2' <- instantiateFull s2'
 
   --Jesper, 2018-04-23: disabled PTS constraints for now,
   --this assumes that piSort can only be blocked by unsolved metas.
+  --Arthur Adjedj, 2023-02-27, Turned PTS back on,
+  --piSort can now be blocked by Leveluniv
+  case piSort' (unEl <$> a) s1' s2' of
+    Right s -> return s
+    Left b -> do
+      let b' = unblockOnAllMetasIn (s1' ,unAbs s2')
+      addConstraint (unblockOnBoth b b') $ HasPTSRule a s2'
+      return $ PiSort (unEl <$> a) s1' s2'
 
-  --case piSort' s1 s2 of
-  --  Just s -> return s
-  --  Nothing -> do
-  --    addConstraint $ HasPTSRule s1 s2
-  --    return $ PiSort s1 s2
-
-  return $ piSort (unEl <$> a) s1' s2'
 
 -- | As @inferPiSort@, but for a nondependent function type.
 inferFunSort :: Sort -> Sort -> TCM Sort
-inferFunSort s1 s2 = funSort <$> reduce s1 <*> reduce s2
+inferFunSort s1 s2 = do
+  ss1 <- sortOf (Sort s1)
+  s1' <- instantiateFull =<< reduce s1
+  s2' <- instantiateFull =<< reduce s2
```

Comment:
We generally try to avoid using `instantiateFull` since it may lead to very big terms, so if it is necessary here there should be a good justification.